src/Microsoft.VisualStudio.Threading.Analyzers/VSTHRD100AsyncVoidMethodCodeFix.cs

```diff
@@ -75,7 +75,7 @@ internal VoidToTaskCodeAction(Document document, Diagnostic diagnostic)
             public override string Title => Strings.VSTHRD100_CodeFix_Title;
 
             /// <inheritdoc />
-            public override string EquivalenceKey => VSTHRD100AsyncVoidMethodAnalyzer.Descriptor.Id;
+            public override string EquivalenceKey => null;
```

Comment:
so you don't want fix all for this fix?

Reply:
Sure I do. And it works. @mavasani's doc says:

> Equivalence Key: A string value representing an equivalence class of all code actions registered by a code fixer or refactoring. Two code actions are treated as equivalent if they have equal non-null EquivalenceKey values and were generated by the same code fixer or refactoring.

But it's inaccurate. At least, it reads to me that a null equivalence key means it won't be batched. But I just tested it, and it works fine.

Reply:
@mavasani seems a bug?

Reply:
"Fixing" that bug (if that's what it is) would break existing code fix providers that intend to be batched and use null, FWIW.

Reply:
This is by design. The default behavior for CodeAction is to have null equivalence key, and this makes the common user scenario of registering a single code fix and not providing an explicit equivalence key to work fine. The analyzer diagnostic is intended to advice providing an `explicit` non-null equivalence key for clarity and maintenance, so if in future you extend your fixer and register another code action, then you don't regress the FixAll for prior registered action.